### ISSUE TITLE
[BUG] Analysis fails with errors when the “Analyze with Open Source Libraries” option is enabled

### ISSUE THREAD

**Issue by sshveta** [konveyor/analyzer-lsp#1078]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

upstream

### Priority

Critical

### Current Behavior

Analysis fails with errors when the “Analyze with Open Source Libraries” option is enabled, but completes successfully when the option is disabled.

Description:
When running an analysis with the Analyze with Open Source Libraries setting selected, the process does not complete successfully and ends with errors. The same analysis configuration runs to completion without any errors when this option is turned off, indicating the issue is specifically related to the open-source library analysis path.

Actual Result:
Analysis fails only when Analyze with Open Source Libraries is enabled.

<img width="2107" height="1096" alt="Image" src="https://github.com/user-attachments/assets/d301cb7c-a3cc-4e06-abb2-d50c0d83bca4" />

### Expected Behavior

Analysis should complete successfully regardless of whether the Analyze with Open Source Libraries option is enabled.

### How Reproducible

Always (Default)

### Steps To Reproduce

- Create two analysis profile, one with (open source libraries) OSL and one without .
- Create two applications and Analyze one with Open Source Libraries
- and second without OSL profile.
- Observe that the analysis ends with errors (Completed with errors)
- Repeat the analysis with the option disabled
- Observe that the analysis completes successfully

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

### Anything else?

Cluster : https://tackle-konveyor-tackle.apps.mtv02.rhos-psi.cnv-qe.rhood.us/ 

[konveyor-analyzer.output(2).txt](https://github.com/user-attachments/files/24870097/konveyor-analyzer.output.2.txt)

**Comment by sshveta** [comment]
Mode of analysis : Source + dependencies .
Analysis was triggered from UI . 
Not tested from analyzer.

**Comment by dymurray** [comment]
After more investigation, this seems to be related to a rule change that is using more memory consumption than prior runs. The solution to this is to bump the memory/cpu requests when running analysis.

There is an investigation into the rule to consider some optimizations but this is not a blocker for 0.9.0

**Comment by shawn-hurley** [comment]
Checking in on if there is a change that needs to be made in the analyzer or if we understand what is happening in the rule change to cause this, let me know if there is anything that we need to do.

**Comment by eemcmullan** [comment]
@shawn-hurley we may need to bump the timeout in the analyzer; maybe only if we are analyzing OS libraries?. It seems expected to take longer in this case:

```
    - java.referenced:
        location: ANNOTATION
        pattern: org.springframework.jmx.*
```
 wdyt?
